opm/input/eclipse/Schedule/Network/ExtNetwork.cpp

```diff
@@ -235,5 +239,29 @@ std::vector<std::string> ExtNetwork::node_names() const
 {
     return this->insert_indexed_node_names;
 }
+
+std::set<std::string> ExtNetwork::leaf_nodes() const
```

Comment:
The function looks correct to me. From the usage in the opm-simulators, it looks like it will be called many times.  Ideally, it will be nice if we can save the results somehow to avoid repeating the process.

This can be done in a following up PR if it turns out to be costly and needs to be optimized.